pages/architecture.rst

```diff
@@ -0,0 +1,224 @@
+required consent record.
+
+Consent authorisation
+---------------------
+
+The consent check is basically the authorization check for the request. Every consent record basically consists of a
+triple: (care provider, patient, care professional), or 'May the care professional access the data for the patient
+managed by the care provider?' The patient can be acquired by analysing the request, for example within FHIR the BSN
+will be included in the request, the care professional can be determined by the attributes used to generate the
+signature. The query to the consent registry within Nuts will then result in a list of care providers which can be used
+to query the correct data (or not found).
+
+Data
+----
+
+If all previous steps have succeeded, the API component can then proceed to fetch and convert the data as requested. The
+initial versions will probably use FHIR as standard of choice.
+
+Patients DB
+===========
+
+This *database* represents the patient's personal information and specifically the BSN record. All EHRs will contain this
+information since it's required by law. The data needs to be accessible by the *patient callback* component.
+
+Patient callback
+================
+
+This component will mainly be used to check if a patient is receiving care for the given care provider. If, for example,
+this care provider receives consent to access certain patient records from another care provider it can only accept this
+consent is the patient is really a patient there. If not, the BSN may not be stored and the consent request can not be
+accepted. This check can also be used to detect faulty or corrupt Nuts nodes, since a lot of negative results from this
+component may indicate fraud. In a later version of Nuts this can be used for automatic blacklisting of Nuts nodes.
+
+EHR UI
+======
+
+The EHR UI represents the piece of software the user interacts with. The part that is particular interesting is the
+consent UI. In the early stages of Nuts, the care providers will probably do all the work for gathering the patient
+consent. This means that the EHR needs to have a UI capable of recording this consent.
+
+PGO UI
+======
+
+This component represents the UI needed for the PGO-inclusion flow. An idea exists where a patient is redirected by a
+PGO to this component to link their PGO identity to a BSN. The vendor can then use the Nuts network to update the
+consent records with the added PGO identity for all existing consent records for that patient. The UI needs to be in
+vendor or service space, otherwise the BSN can not be used. The difference between putting it in vendor or service space
+would be if it's embedded or not. Nuts will provide a reference implemention for placing it in the *service space*.
+
+*************
+Service space
+*************
+
+*Service space* is a bit of a difficult concept to explain. The choice to implement/host this by the vendor is based on
```

Comment:
Maybe it can be made easier when you explain the concepts of levels of trust earlier in the document?

Reply:
Moved to beginning